qa327/frontend.py

```diff
@@ -28,11 +41,13 @@ def register_post():
     if password != password2:
         error_message = "The passwords do not match"
 
-    elif len(email) < 1:
-        error_message = "Email format error"
+    elif len(email) < 1 or not EMAIL_REGEX.match(email):
+        error_message = "Email format is incorrect"
 
-    elif len(password) < 1:
-        error_message = "Password not strong enough"
+    elif len(password) < 6 or not (any(x.isupper() for x in password)) or not (any(x.islower() for x in password)) or not (any(not x.isalnum() for x in password)):
+        error_message = "Password format is incorrect / not strong enough"
+    elif len(name) < 2 or len(name) >= 20 or (any(not x.isalnum() for x in name)) or name[0] == ' ' or name[-1] == ' ':
```

Comment:
I would avoid calling `len` on `name` more than once - maybe store the length in a variable before the `if` statement.